mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java

```diff
@@ -862,6 +882,150 @@ private Future<Errors> validateMappingProfile(OperationType operationType, Mappi
+      return Future.succeededFuture(new Errors().withTotalRecords(0));
+    }
+
+    var recordType = actionProfileUpdateDto.getProfile().getFolioRecord();
+    var errors = new LinkedList<Error>();
+    Promise<Errors> promise = Promise.promise();
+
+    var futures = actionProfileUpdateDto
+      .getAddedRelations()
+      .stream()
+      .filter(profileAssociation -> profileAssociation.getDetailProfileType() == ProfileAssociation.DetailProfileType.MAPPING_PROFILE)
+      .map(profileAssociation -> mappingProfileService.getProfileById(profileAssociation.getDetailProfileId(), false, tenantId))
+      .map(futureMappingProfile -> futureMappingProfile.onSuccess(optionalMappingProfile -> {
+        optionalMappingProfile.ifPresent(mappingProfile -> {
+          if (!Objects.equals(mappingProfile.getExistingRecordType().value(), recordType.value())) {
+            logger.info("Can not update ActionProfile with ID:{} because FolioRecord:{}, linked MappingProfile FolioRecord:{}",
+              actionProfileUpdateDto.getProfile().getId(), recordType.value(), mappingProfile.getExistingRecordType().value());
+            errors.add(new Error().withMessage(String.format(INVALID_RECORD_TYPE_LINKED_MAPPING_PROFILE_TO_ACTION_PROFILE, mappingProfile.getName())));
+          }
+        });
+      }))
+      .collect(Collectors.toList());
+    GenericCompositeFuture.all(futures)
+      .onSuccess(handler -> promise.complete(new Errors().withErrors(errors)))
+      .onFailure(promise::fail);
+
+    return promise.future();
+  }
+
+  private Future<Errors> validateActionProfileChildProfilesFolioRecord(ActionProfileUpdateDto actionProfileUpdateDto, String tenantId, String id) {
+    var recordType = actionProfileUpdateDto.getProfile().getFolioRecord();
+    var errors = new LinkedList<Error>();
+    var deletedRelations = actionProfileUpdateDto.getDeletedRelations();
+    Promise<Errors> promise = Promise.promise();
+
+    actionProfileService.getProfileById(id, true, tenantId)
+      .onSuccess(optionalActionProfile ->
+        optionalActionProfile.ifPresentOrElse(actionProfile -> {
+            var existMappingProfiles = CollectionUtils.isEmpty(deletedRelations) ? actionProfile.getChildProfiles() :
+              actionProfile.getChildProfiles().stream()
+                .filter(profileSnapshotWrapper -> profileSnapshotWrapper.getContentType() == ContentType.MAPPING_PROFILE)
+                .filter(profileSnapshotWrapper -> {
+                  for (ProfileAssociation deletedRelation : deletedRelations) {
+                    if (Objects.equals(deletedRelation.getDetailProfileId(), profileSnapshotWrapper.getProfileId())) {
+                      return false;
+                    }
+                  }
+                  return true;
+                }).collect(Collectors.toList());
+
+            existMappingProfiles.forEach(mappingProfile -> {
+              if (!Objects.equals(actionProfile.getFolioRecord().value(), recordType.value())) {
```

Comment:
we should use here the mappingProfile variable instead of actionProfile to compare record types, right?